qutip/solver/heom/bofin_solvers.py

```diff
@@ -726,6 +752,39 @@ def _combine_bath_exponents(self, bath):
             )
         return exponents
 
+    def _is_environment_api(self, bath_spec):
+        is_list = isinstance(bath_spec, (list, tuple))
+        if not is_list:
+            return False
+
+        starts_with_env = isinstance(
+            bath_spec[0], (BosonicEnvironment, FermionicEnvironment)
+        )
+        if not starts_with_env:
+            return False
+
+        contains_bath = any(
+            isinstance(elem, (Bath, BosonicBath, FermionicBath))
+            for elem in bath_spec
+        )
+        if contains_bath:
+            # Potentially confused user, passed mixed list of environments
+            # and baths. Error will be raised elsewhere.
+            return False
+
+        return True
```

Comment:
```suggestion
        if not isinstance(bath_spec, (list, tuple)) or len(bath_spec) != 2:
            return False
        env, Q = bath_spec

        if not isinstance(env, (BosonicEnvironment, FermionicEnvironment)):
            return False

        if not isinstance(q, (Qobj, QobjEvo):
            return False

        return True
```
I'm not sure if there needs to be a slightly more complex check for whether the `env` is the right kind of bath or not, but the length and ordering of the expected pair should definitely be checked, yes?